FoodOrderingSystem.py

```diff
@@ -0,0 +1,813 @@
+        """ Cancel order """
+
+        update = session.query(
+            CustOrderStatus
+            ).filter(CustOrderStatus.order_id == order_id
+            ).update(
+                {CustOrderStatus.order_status: order_status}, 
+                synchronize_session=False
+                )
+        try:
+            session.commit()
+            return update
+        except:
+            session.rollback()
+            raise Exception("Checkout not completed!")
+        finally:
+            session.expunge_all()
+            session.close()
+
+    def view_orders_status(self, order_id, session):
+        """ Customer can view their orders after checkout/confirming order """
+
+        try:
+            view = session.query(FoodCategory, FoodDetails, CustOrderSelection, CustomerDetails, CustOrderStatus, DeliveryPerson, func.sum(CustOrderSelection.food_qty*FoodDetails.price)).\
+                filter(CustomerDetails.cust_id == CustOrderStatus.cust_id).\
+                filter(CustOrderSelection.food_id == FoodDetails.food_id).\
+                filter(FoodCategory.category_id == FoodDetails.category_id).\
+                filter(CustOrderSelection.order_id == CustOrderStatus.order_id).\
+                filter(DeliveryPerson.delivery_person_id == CustOrderStatus.delivery_person_id).\
+                filter(CustOrderStatus.order_id == order_id)
+            return view
+        except Exception as ex:
+            print("Error getting order, error={}".format(str(ex)))
+        finally:
+            session.close()
+
+class Controller(SQLiteBackend):
+    """ Controller class that inherites from SQLite Backend and composition from Emploeyee, Customer and Delivery Person classes """
+
+    def __init__(self, db_url):
+        super(Controller, self).__init__(db_url)
+        self.employee = Employee()
+        self.customer = Customer()
+        self.delivery_person = DeliveryPerson()
+
+class DeliveryPerson(Base, Employee):
+    """ Represents delivery person """
+
+    __tablename__ = 'delivery_person'
+
+    delivery_person_id = Column(Integer(), primary_key=True, nullable=False, unique=True, autoincrement=True)
+    delivery_person_name = Column(String(30), nullable=False)
+    delivery_person_phone = Column(Integer(), nullable=False)
+
+    def update_order(self, order_id, order_status, session):
+        """ Delivery person can update the order """
+
+        super(DeliveryPerson, self).__init__()
+
+        update = session.query(
+            CustOrderStatus
+            ).filter(CustOrderStatus.order_id == order_id
+            ).update(
+                {CustOrderStatus.order_status: order_status}, 
+                synchronize_session=False
+                )
+        try:
+            session.commit()
+            return update
+        except:
+            session.rollback()
+            raise Exception("update not completed!")
+        finally:
+            session.expunge_all()
+            session.close()
+
+def main():
```

Comment:
The main() function is so huge and complicated. It needs to be refactored and simplified. Break down the logic into smaller functions as part of the Controller class and call it from the main. For example:

```python

class Controller(SQLiteBackend):

    def add_food_category(...):
        ...
        ...


def main():
    
    ...
    ...
    fos = Controller(db_url)
    fos.bootstrap()
    ...
    ...

    while employee_options != 0:
        if employee_options == 1:
            fos.add_food_category(...)
    ...
    ...
```

Reply:
Refactored def add_food_category()

    def add_food_category(self, category_name, session):
        c = self.employee.add_food_category(category_name, session)
        if c:
            print("\nCategory ID: {} \nCategory Name: {} \nSuccessfully!".format(c.category_id, c.name))
        else:
            print("Adding details was not successfully. Please try again!")    

        while employee_options != 0:

            if employee_options == 1:
                category_name = input("Enter category name: ")
                session = fos.Session()
                fos.add_food_category(category_name, session)